lib/AST/Type.cpp

```diff
@@ -4197,6 +4197,10 @@ Type Type::subst(TypeSubstitutionFn substitutions,
 }
 
 Type Type::substDependentTypesWithErrorTypes() const {
+  if (!getPointer()->hasTypeParameter()) {
```

Comment:
Can the type have archetypes here too?

Reply:
IIRC substDependentTypesWithErrorTypes() was a hack and it would be better to let mapTypeIntoContext() crash on an invalid type and fix callers to not pass in invalid types instead. We should try ripping it out at some point and see what breaks.